### DISCUSSION TITLE
Work with Angular Ionic PWA and Native

### DISCUSSION THREAD

**Discussion by GVAMDaniel** [AR-js-org/AR.js#583] [Help]
I managed to make it work using an HTML located within the assets that performs all the AR.js functionalities.

To call it from Angular, I used an iframe like this: 
<iframe #iframe src="./assets/ar/ar.html"></iframe>
and for communication between the HTML and Angular, I used an @HostListener('window:message', ['$event'])

All of this worked perfectly, for example, when using it in an Ionic application on iOS or Web as PWA.
There has a problem has arisen with the requirement of Android 13 SDK 33.
I'm not sure what has happened, but when running an Ionic app on Android, it no longer detects the camera, and an error is raised, stating that camera permission has not been granted, despite using Capacitor plugins that indicate it should be working.

Has anyone who has done something similar managed to get AR.js to work on Android?

**Comment by Rossella-Mascia-Neosyn** [comment]
@GVAMDaniel Did you get it to work?

**Reply by GVAMDaniel** [reply]
Nop sorry, we abandon the project and migrate everything to flutter.

Anyway, it is possible to create a plugin that works with ionic using native ArCore or ArKit, the problem is the time to develop it with the specifications you need.
